pkg/services/accesscontrol/database/database.go

```diff
@@ -277,7 +278,12 @@ func (s *AccessControlStore) SearchUsersPermissions(ctx context.Context, orgID i
 }
 
 // GetUsersBasicRoles returns the list of user basic roles (Admin, Editor, Viewer, Grafana Admin) indexed by UserID
-func (s *AccessControlStore) GetUsersBasicRoles(ctx context.Context, userFilter []int64, orgID int64) (map[int64][]string, error) {
+// Special case for anonymous users, they don't have a user ID, we return the basic roles for them
+func (s *AccessControlStore) GetUsersBasicRoles(ctx context.Context, userIDFilter []int64, orgID int64) (map[int64][]string, error) {
+	// Special case for anonymous users, they don't have a user ID
+	if len(userIDFilter) == 1 && userIDFilter[0] == 0 {
+		return map[int64][]string{0: []string{string(identity.RoleAdmin), string(identity.RoleEditor), string(identity.RoleViewer)}}, nil
+	}
```

Comment:
question: isn't this logic assigning all roles to the anonymous user? Don't we want to look `AnonymousOrgRole` and `AnonymousOrgName` to return anonymous permissions in the given org?

Reply:
Yeah, this doesn't look right

Reply:
my current logic looks like:

```go
	var err error
	var usersRoles map[int64][]string
	if usr.IsIdentityType(claims.TypeAnonymous) {
		usersRoles = map[int64][]string{0: {string(usr.GetOrgRole())}}
	} else {
		usersRoles, err = s.store.GetUsersBasicRoles(ctx, nil, usr.GetOrgID())
		if err != nil {
			return nil, err
		}
	}
```

@IevaVasiljeva you are right we were giving the anonymoususer all the roles.

using the `identity.Requester` we can get the OrgRole from `GetOrgRole()`.

in the case for 636 inside of `searchUserPermissions` we have another case where we would use:
```go
	var roleList map[int64][]string
	// Special case for anonymous users, they don't have a user ID
	if userID == 0 {
		roleList = map[int64][]string{0: {string(s.cfg.AnonymousOrgRole)}}
	} else {
		roleList, err = s.store.GetUsersBasicRoles(ctx, []int64{userID}, orgID)
		if err != nil {
			return nil, fmt.Errorf("could not fetch basic roles for the user: %w", err)
		}
	}
```

this is because we do not have access to the usr.Identity.Requester interface. Does this make sense?

Reply:
I think that makes sense, yes. Will review more carefully once you have pushed the new changes.